core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -611,13 +611,21 @@ object Parser extends ParserInstances {
   def pure[A](a: A): Parser[A] =
     Impl.Pure(a)
 
+  /** Parse a given string, in a case-insensitive manner,
+    * or fail. This backtracks on failure
+    * this is an error if the string is empty
+    */
+  def stringCI1(str: String): Parser1[Unit] =
+    if (str.length == 1) charIn(str.charAt(0).toLower, str.charAt(0).toUpper).void
```

Comment:
nit, can we call charAt once and capture in a val? It makes it a bit easier to read.